BuddySave.IntegrationTests/CloudManagerTests.cs

```diff
@@ -0,0 +1,287 @@
+            gameSave.CloudPath = cloudSave.Path;
+            gameSave.LocalPath = localSave.Path;
+
+            cloudSave.Create();
+            localSave.Create();
+
+            var oldSavePath = Path.Combine(cloudSave.Path, oldSaveFileName);
+            await File.WriteAllTextAsync(oldSavePath, "Test");
+
+            // Act
+            sut.UploadSave(gameSave);
+
+            // Assert
+            Assert.False(Directory.Exists(oldSavePath));
+        }
+
+        [Theory, AutoMoqData]
+        public void UploadSave_CopiesEmptyLocalToCloud_WhenCloudSaveDoesNotExist(
+            GameSave gameSave,
+            CloudManager sut)
+        {
+            // Arrange
+            using var cloudSave = new TempSaveDir(gameSave.CloudPath);
+            using var localSave = new TempSaveDir(gameSave.LocalPath);
+            gameSave.CloudPath = cloudSave.Path;
+            gameSave.LocalPath = localSave.Path;
+
+            localSave.Create();
+
+            // Act
+            sut.UploadSave(gameSave);
+
+            // Assert
+            Assert.True(Directory.Exists(gameSave.CloudPath));
+        }
+
+        [Theory, AutoMoqData]
+        public async Task UploadSave_CopiesFilesToCloud_WhenLocalHasFiles(
+            string file,
+            GameSave gameSave,
+            CloudManager sut)
+        {
+            // Arrange
+            using var cloudSave = new TempSaveDir(gameSave.CloudPath);
+            using var localSave = new TempSaveDir(gameSave.LocalPath);
+            gameSave.CloudPath = cloudSave.Path;
+            gameSave.LocalPath = localSave.Path;
+
+            localSave.Create();
+            var localFilePath = Path.Combine(localSave.Path, file);
+            var cloudFilePath = Path.Combine(cloudSave.Path, file);
+            await File.WriteAllTextAsync(localFilePath, "Test");
+
+            // Act
+            sut.UploadSave(gameSave);
+
+            // Assert
+            Assert.True(File.Exists(cloudFilePath));
+        }
+
+        [Theory, AutoMoqData]
+        public async Task DownloadSave_DeletesOldLocalSave_WhenLocalSaveExists(
+            string oldSaveFileName,
+            GameSave gameSave,
+            CloudManager sut)
+        {
+            // Arrange
+            using var localSave = new TempSaveDir(gameSave.LocalPath);
+            using var cloudSave = new TempSaveDir(gameSave.CloudPath);
+            gameSave.LocalPath = localSave.Path;
+            gameSave.CloudPath = cloudSave.Path;
+
+            localSave.Create();
+            cloudSave.Create();
+
+            var oldSavePath = Path.Combine(localSave.Path, oldSaveFileName);
+            await File.WriteAllTextAsync(oldSavePath, "Test");
+
+            // Act
+            sut.DownloadSave(gameSave);
+
+            // Assert
+            Assert.False(Directory.Exists(oldSavePath));
+        }
+
+        [Theory, AutoMoqData]
+        public void DownloadSave_CopiesEmptyCloudToLocal_WhenLocalSaveDoesNotExist(
```

Comment:
I think I'm starting to get what `CopiesEmpty` mean - it creates an empty directory in destination? Wouldn't this essentially delete the cloud/local files if local/cloud dir was empty/did not exist?

Reply:
Yes, we delete the saves before copying either way. Was thinking this would give better predictability.

This could cause issues with config failures, I guess 🤔